s3po/backends/memory.py

```diff
@@ -19,7 +19,7 @@ def download(self, bucket, key, fobj, retries, headers=None):
         else:
             fobj.write(obj)
 
-    def upload(self, bucket, key, fobj, retries, headers=None):
+    def upload(self, bucket, key, fobj, retries, headers=None, extra=None):
```

Comment:
I thought `headers` were part of `extra` now?

Reply:
@b4hand

Reply:
Yes, for S3. However, this interface is maintained as common for the Swift and "memory" backends. Unfortunately, with this change the headers aren't being transferred the same for Swift and S3, so code can't really be common. I could have written another compatibility layer, but ultimately our usage of Swift has essentially dropped off to nothing, and there's other differences between Swift and S3 that make it impossible to use the same code for both, so in reality, it was mostly a pipe dream to maintain compatibility between both Swift and S3.